.circleci/config.yml

```diff
@@ -0,0 +1,130 @@
+    working_directory: ~/project/build
+    docker:
+      - image: circleci/node:8
+
+  workspace_root: &workspace_root
+    ~/project
+
+  attach_workspace: &attach_workspace
+    attach_workspace:
+      at: *workspace_root
+
+  npm_cache_keys: &npm_cache_keys
+    keys:
+      - v1-dependency-npm-{{ checksum "package.json" }}
+      - v1-dependency-npm-
+
+  cache_npm_cache: &cache_npm_cache
+    save_cache:
+      key: v1-dependency-npm-{{ checksum "package.json" }}
+      paths:
+        - ./node_modules/
+        - ./tools/x-docs/node_modules/
+        - ./tools/x-workbench/node_modules/
+
+  restore_npm_cache: &restore_npm_cache
+    restore_cache:
+      <<: *npm_cache_keys
+
+  filters_ignore_tags: &filters_ignore_tags
+    tags:
+      ignore: /.*/
+
+  filters_version_tag: &filters_version_tag
+    tags:
+      only:
+        - /^v\d+\.\d+\.\d+(?:-beta\.\d+)?$/
+    branches:
+      ignore: /.*/
+
+jobs:
+
+  build:
+    <<: *container_config_node8
+    steps:
+      - checkout
+      - run:
+          name: Checkout next-ci-shared-helpers
+          command: git clone --depth 1 git@github.com:Financial-Times/next-ci-shared-helpers.git .circleci/shared-helpers
+      - *restore_npm_cache
+      - run:
+          name: Install project dependencies
+          command: make install
+      - run:
+          name: Run the project build task
+          command: make build
+      - *cache_npm_cache
+      - persist_to_workspace:
+          root: *workspace_root
+          paths:
+            - build
+
+  test:
+    <<: *container_config_node8
+    steps:
+      - *attach_workspace
+      - run:
+          name: Run tests
+          command: make test
+      #    environment:
+      #       JEST_JUNIT_OUTPUT: test-results/jest/results.xml
+      # - store_test_results:
+      #     path: test-results
+      # - store_artifacts:
+      #     path: test-results
+      #     destination: test-results
+
+  publish:
+    <<: *container_config_node8
+    steps:
+      - *attach_workspace
+      - run:
+          name: shared-helper / npm-store-auth-token
+          command: .circleci/shared-helpers/helper-npm-store-auth-token
+      - run:
+          name: NPM publish
+          command: npx athloi version ${CIRCLE_TAG} && npx athloi publish -- --access=public
+      - run:
+          name: Publish GitHub Pages
+          command: ./private/scripts/gh-pages
+
+workflows:
+
+  version: 2
+
+  build-test:
+    jobs:
+      - build:
+          filters:
+            <<: *filters_ignore_tags
+      - test:
+          filters:
+            <<: *filters_ignore_tags
+          requires:
+            - build
+
+  build-test-publish-deploy:
+    jobs:
+      - build:
+          filters:
+            <<: *filters_version_tag
+      - test:
+          filters:
+            <<: *filters_version_tag
+          requires:
+            - build
+      - publish:
+          filters:
+            <<: *filters_version_tag
+          requires:
+            - test
+
```

Comment:
Possibly worth mentioning that you won't get a nightly build unless you specify a scheduled workflow here i.e. `next-rebuild-bot` isn't triggering builds for projects that are on CircleCI 2.0.

Reply:
:ok_hand: